doc/dev/error-message.md

```diff
@@ -0,0 +1,14 @@
+# Error message
+
+Elektra has a lot of developers and hence the error messages vary in how they are written.
+This guideline should unify the message format so that users will get consistent error messages.
+
+- All error messages start with a capital letter even if you reference a method name such as "KdbSet() not supported". The only exception are quoted messages at the start such as "'xxxx-xxxx' is an invalid syntax". Try to avoid variable strings at the start of the sentence ("%s"-strings) if possible because you cannot guarantee capital casing.
+- **Always** provide the errno if possible.
+- If you can provide an errno, write `"Reason: %s", strerror(errno)` at the end of the error message. Also end the preceding sentence with a dot. Example: `"XY failed. Reason: %s", strerror(errno)`. Alternatively you can also embed it into a standard text: `"XY failed because of %s", strerror(errno)`.
```

Comment:
We should not recommend to call strerror(errno) as it is not thread-safe. strerror_r should be used instead. It would be better if we can embed the strerror in a generic way, so that not every caller needs to worry about that.

Reply:
> It would be better if we can embed the strerror in a generic way, so that not every caller needs to worry about that.

This is already written in https://github.com/ElektraInitiative/libelektra/issues/2572

Reply:
> We should not recommend to call strerror(errno) as it is not thread-safe.

This seems a bit problematic: http://www.club.cc.cmu.edu/~cmccabe/blog_strerror.html
Always allocating and freeing memory just for an error message seems very cumbersome.
Also what you mentioned multiple times now is that elektra is not intended to be used with multi threading. So why now introduce thread safety just for error messages?